kamon-play-2.6.x/src/main/scala/kamon/play/instrumentation/RequestHandlerInstrumentation.scala

```diff
@@ -15,48 +15,54 @@
 
 package kamon.play.instrumentation
 
-import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
 import kamon.Kamon
 import kamon.context.Context
-import kamon.play.OperationNameFilter
 import kamon.trace.Span
 import kamon.util.CallingThreadExecutionContext
-import org.aspectj.lang.ProceedingJoinPoint
-import org.aspectj.lang.annotation._
-import play.api.mvc.EssentialFilter
 
 import scala.concurrent.Future
 
-@Aspect
-class RequestHandlerInstrumentation {
+trait GenericRequest {
+  val headers: Map[String, String]
```

Comment:
I would like to know what you guys think about this on the context of https://github.com/kamon-io/kamon-play/pull/21. As I mentioned there, it might be simpler to just have a `getHeader(...)` method and let the underlying HTTP model take care of retrieving headers properly.

Reply:
How about using `play.api.mvc.Headers` instead of `Map[String, String]`, which will provide exactly the case-insensitive functionality needed there, as well as relevant helpers.

https://github.com/playframework/playframework/blob/386b52aec425a85234ad3c3c5429c2002642bc66/framework/src/play/src/main/scala/play/api/mvc/Headers.scala#L22

Reply:
I like the idea!!

Reply:
Sorry, I don't think that quite makes sense. For the internal code here, it may make sense to use `play.api.mvc.Headers`, but in #21, the point is to populate a public trait `TextMap`, which is broader scope than Play. I wonder if there needs to be some thought into making `TextMap` itself act more like `play.api.mvc.Headers` in terms of case-insensitivity, multiple values per key, etc?

Reply:
My main argument against using `play.api.mvc.Headers` or a map of our own is the unnecessary conversions between the HTTP Request model and the Map/Headers object and the garbage generated in the process. The Context codecs are meant to know exactly which headers to look for and a simple API that allows to get by header name should be more than enough.

(btw, take a look at https://github.com/kamon-io/Kamon/issues/526 which is somehow related)